### PR TITLE
Invoke bazel build during sync using --target_pattern_fi…

### PR DESCRIPTION
**Pull request by sfc-gh-mgalindo**
Use --target_pattern_file flag during sync instead of passing list of targets to avoid large command line errors in the same spirit as https://github.com/bazelbuild/intellij/pull/6537.

### CODE REVIEW

**Comment by tpasternak** [comment]
@sfc-gh-mgalindo it's still in draft mode, would you like us to review it?

**Comment by sfc-gh-mgalindo** [comment]
> @sfc-gh-mgalindo it's still in draft mode, would you like us to review it?

@tpasternak  yes, please review. Marked it as ready for review now

File: base/src/com/google/idea/blaze/base/sync/aspects/BlazeIdeInterfaceAspectsImpl.java [hunk 1]
```diff
@@ -763,6 +772,36 @@ private static BlazeBuildOutputs runBuildForTargets(
...
+              logger.error("Failed to delete target pattern file", e);
+          }
+      }
+    }
+  }
+
+  private static @NotNull Path prepareTargetPatternFile(List<? extends TargetExpression> targets) throws BuildException {
+    Path targetPatternFile = null;
+    try {
+      targetPatternFile = Files.createTempFile("targets-", "");
```
Comment (mai93) [inline base/src/com/google/idea/blaze/base/sync/aspects/BlazeIdeInterfaceAspectsImpl.java:789]: can we create the file in a directory under the project similar to here: https://github.com/tpasternak/bazel-intellij/blob/769b653fa5ca7a71f92c44f897584e1d3b2011de/base/src/com/google/idea/blaze/base/dependencies/BlazeQuerySourceToTargetProvider.java#L229?

**Comment by tpasternak** [comment]
@sfc-gh-mgalindo  Thank you for the contribution! I agree with Mai, could you please put the temporary in the same directory as the previous one? Apart from that LGTM

**Comment by sfc-gh-mgalindo** [comment]
> @sfc-gh-mgalindo Thank you for the contribution! I agree with Mai, could you please put the temporary in the same directory as the previous one? Apart from that LGTM

Done.

**Comment by tpasternak** [comment]
Thanks, please only fix the compilation errors and we'll merge it

**Comment by sfc-gh-mgalindo** [comment]
> Thanks, please only fix the compilation errors and we'll merge it

Thanks!

**Review by tpasternak — APPROVED** [review]

**Comment by tpasternak** [comment]
@satyanandak may I ask you to rerun the CI (one of the test suites did not start, so I can't merge the PR)

**Pull request merged**